base/ca/src/com/netscape/cms/servlet/cert/scep/CRSEnrollment.java

```diff
@@ -404,6 +414,31 @@ public void service(HttpServletRequest httpReq,
 
     }
 
+    /*
+     * If this is a POST request, try reading the binary message from the request body.
+     */
+    private String readMessageFromPostBody(HttpServletRequest httpReq) {
+        String message = null;
+
+        try {
+            if (httpReq.getMethod().equalsIgnoreCase("POST")) {
+                ServletInputStream is = httpReq.getInputStream();
+                ByteArrayOutputStream bstream = new ByteArrayOutputStream(10000);
```

Comment:
Is this an attempt to limit the size of the input data?  If so, I don't think it'd work as the buffer will grow as needed for ByteArrayOutputStream.  Now the question is on whether we want to set a limit on the input data.  Any suggestion, @cipherboy or @edewata ?

Reply:
Yeah, I think this will limit the input data, but since the `try-catch` block surrounding this code will discard the exception, I'm not quite sure what would happen if the limit is exceeded, whether it will fall back to the old behavior or completely fail. IIUC the `getParameter()` cannot be called after `getInputStream()`, so it might completely fail.

Is there a defined limit on the size of a valid SCEP input?

Reply:
Since we pull Commons IO, do we want to use [IOUtil.Copylarge](https://commons.apache.org/proper/commons-io/apidocs/org/apache/commons/io/IOUtils.html)?